materialize-databricks/driver.go

```diff
@@ -147,20 +147,22 @@ func newTransactor(
 
 	var d = &transactor{cfg: cfg, wsClient: wsClient}
 
-	if d.updateDelay, err = m.ParseDelay(cfg.Advanced.UpdateDelay); err != nil {
+	var httpPathSplit = strings.Split(cfg.HTTPPath, "/")
+	var warehouseId = httpPathSplit[len(httpPathSplit)-1]
+
+	if sched, useSched, err := boilerplate.CreateSchedule(cfg.Schedule, []byte(warehouseId), cfg.Advanced.UpdateDelay); err != nil {
 		return nil, err
+	} else if useSched {
+		d.sched = sched
 	}
 
 	// If the warehouse has auto-stop configured to be longer than 15 minutes, disable update delay
-	var httpPathSplit = strings.Split(cfg.HTTPPath, "/")
-	var warehouseId = httpPathSplit[len(httpPathSplit)-1]
 	if res, err := wsClient.Warehouses.GetById(ctx, warehouseId); err != nil {
 		return nil, fmt.Errorf("get warehouse %q details: %w", warehouseId, err)
 	} else {
-		if res.AutoStopMins >= 15 && d.updateDelay.Minutes() > 0 {
+		if res.AutoStopMins >= 15 {
```

Comment:
Yeah this does get more complicated now, because a typical configuration would have the "slow" delay be 4 hours during off-hours, and something else during "prime" hours. So really anything less than a 4 hour auto-stop could technically save costs, depending on how the update schedule is configured.

I'm kind of feeling like we should get rid of this check entirely now and just handle it via documentation. Otherwise we'd need to do a bunch of largely redundant duration parsing of the provided schedule config and/or expose inner workings of the `sched` returned from `boilerplate.CreateSchedule` to be able to do the comparisons in the code and I'm not totally sure how we'd formulate those comparisons. My memory is fuzzy but it may have originally been my suggestion to add a check like this in the first place so I'm happy to remove it now.

Reply:
If the default/typical is now 4 hours during slow time I think it makes sense to remove this 👍🏽